System/Posix/Directory/Common.hsc

```diff
@@ -32,12 +35,24 @@ module System.Posix.Directory.Common (
 import System.Posix.Types
 import Foreign
 import Foreign.C
+import Data.Functor ((<$>))
 
 newtype DirStream = DirStream (Ptr CDir)
 
 data {-# CTYPE "DIR" #-} CDir
 data {-# CTYPE "struct dirent" #-} CDirent
 
+#ifdef HAVE_FDOPENDIR
+-- | @unsafeOpenDirStreamFd fd@ calls @fdopendir@ to obtain a directory stream
+--   for @fd@. @fd@ must not be otherwise used after this; see
+--   <https://pubs.opengroup.org/onlinepubs/9699919799/functions/fdopendir.html POSIX>.
+unsafeOpenDirStreamFd :: Fd -> IO DirStream
+unsafeOpenDirStreamFd (Fd fd) = DirStream <$> throwErrnoIfNull "openDirStreamFd" (c_fdopendir fd)
+
+foreign import capi unsafe "dirent.h fdopendir"
```

Comment:
Let's add a comment saying that the reason using capi is an actual bug.

Reply:
Agreed, we don't want this accidentally reverted back.